core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -47,313 +45,339 @@
+		}
+
+		String maxTransmissionStorageCapacity = namesAndValues.get(MAX_TRANSMISSION_STORAGE_CAPACITY_NAME);
+		initialize(endpointAddress, maxTransmissionStorageCapacity, developerMode, maxTelemetryBufferCapacityEnforcer,
+				sendIntervalInSecondsEnforcer, throttling, maxInstantRetries);
+	}
+
+	/**
+	 * Gets value indicating whether this channel is in developer mode.
+	 */
+	@Override
+	public boolean isDeveloperMode() {
+		return developerMode;
+	}
+
+	/**
+	 * Sets value indicating whether this channel is in developer mode.
+	 *
+	 * @param developerMode
+	 *            True or false
+	 */
+	@Override
+	public void setDeveloperMode(boolean developerMode) {
+		if (developerMode != this.developerMode) {
+			this.developerMode = developerMode;
+			int maxTelemetriesInBatch = this.developerMode ? 1 : DEFAULT_MAX_TELEMETRY_BUFFER_CAPACITY;
+
+			setMaxTelemetriesInBatch(maxTelemetriesInBatch);
+		}
+	}
+
+	/**
+	 * Sends a Telemetry instance through the channel.
+	 */
+	@Override
+	public void send(Telemetry telemetry) {
+		Preconditions.checkNotNull(telemetry, "Telemetry item must be non null");
+
+		if (isDeveloperMode()) {
+			telemetry.getContext().getProperties().put("DeveloperMode", "true");
+		}
+
+		if (telemetrySampler != null) {
+			if (!telemetrySampler.isSampledIn(telemetry)) {
+				return;
+			}
+		}
+
+		StringWriter writer = new StringWriter();
+		JsonTelemetryDataSerializer jsonWriter = null;
+		try {
+			jsonWriter = new JsonTelemetryDataSerializer(writer);
+			telemetry.serialize(jsonWriter);
+			jsonWriter.close();
+			String asJson = writer.toString();
+			telemetryBuffer.add(asJson);
+			telemetry.reset();
+			if (itemsSent.incrementAndGet() % 10000 == 0) {
+				InternalLogger.INSTANCE.info("items sent till now %d", itemsSent.get());
+			}
+
+		} catch (IOException e) {
+			InternalLogger.INSTANCE.error("Failed to serialize Telemetry");
+			InternalLogger.INSTANCE.trace("Stack trace is %s", ExceptionUtils.getStackTrace(e));
+			return;
+		}
+
+		if (isDeveloperMode()) {
+			writeTelemetryToDebugOutput(telemetry);
+		}
+	}
+
+	/**
+	 * Stops on going work
+	 */
+	@Override
+	public synchronized void stop(long timeout, TimeUnit timeUnit) {
+		try {
+			if (stopped) {
+				return;
+			}
+
+			telemetriesTransmitter.stop(timeout, timeUnit);
+			stopped = true;
+		} catch (Throwable t) {
+			InternalLogger.INSTANCE.error("Exception generated while stopping telemetry transmitter");
+			InternalLogger.INSTANCE.trace("Stack trace generated is %s", ExceptionUtils.getStackTrace(t));
+		}
+	}
+
+	/**
+	 * Flushes the data that the channel might have internally.
+	 */
+	@Override
+	public void flush() {
+		telemetryBuffer.flush();
+	}
+
+	/**
+	 * Sets an optional Sampler that can sample out telemetries Currently, we don't
+	 * allow to replace a valid telemtry sampler.
+	 *
+	 * @param telemetrySampler
+	 *            - The sampler
+	 */
+	@Override
+	public void setSampler(TelemetrySampler telemetrySampler) {
```

Comment:
This should be deprecated. It is the old sampling strategy which we do not encourage any more. If we are not scrapping it off entirely then atleast please use @depracted tag here

Reply:
@dhaval24 - same here. Unrelated to this PR.